tgui/packages/tgui/interfaces/EventPanel.js

```diff
@@ -0,0 +1,155 @@
+                          label={vote.name}>
+                          {vote.votes}
+                        </LabeledList.Item>
+                      ))}
+                    </LabeledList>
+                  ) : (
+                    <NoticeBox>
+                      No vote in progress.
+                    </NoticeBox>
+                  )}
+                </Section>
+              </Stack.Item>
+            </>
+          ) : (
+            !!show_votes && (
+              <Stack.Item>
+                <Section title="Current Votes">
+                  {vote_in_progress ? (
+                    <LabeledList>
+                      {votes.map(vote => (
+                        <LabeledList.Item
+                          key={vote.id}
+                          label={vote.name}>
+                          {vote.votes}
+                        </LabeledList.Item>
+                      ))}
+                    </LabeledList>
+                  ) : (
+                    <NoticeBox>
+                      No vote in progress.
+                    </NoticeBox>
+                  )}
+                </Section>
+              </Stack.Item>
+            )
+          )}
+          <Stack.Item grow>
+            <Section scrollable fill grow title={vote_in_progress ? "Available Events (" + end_time + " seconds) " : "Available Events"}>
+              {vote_in_progress ? (
+                <LabeledList>
+                  {event_list.map(event => (
+                    <LabeledList.Item
+                      label={event.name}
+                      key={event.name}
+                      buttons={(
+                        <Button
+                          color={event.self_vote ? "good" : "blue"}
+                          icon="vote-yea"
+                          content="Vote"
+                          onClick={() => act('register_vote', {
+                            event_ref: event.ref,
+                          })} />
+                      )} />
+                  ))}
+                </LabeledList>
+              ) : (
+                <NoticeBox>
+                  No vote in progress.
+                </NoticeBox>
+              )}
+            </Section>
+          </Stack.Item>
+          {!!admin_mode && (
+            <Stack.Item>
+              <Section scrollable grow fill height="150px" title="Previous Events">
+                {previous_events.length > 0 ? (
+                  <LabeledList>
+                    {previous_events.map(event => (
+                      <LabeledList.Item
+                        label="Event"
+                        key={event}>
+                        {event}
+                      </LabeledList.Item>
+                    ))}
+                  </LabeledList>
+                ) : (
+                  <NoticeBox>
+                    No previous events.
+                  </NoticeBox>
+                )}
+              </Section>
+            </Stack.Item>
+          )}
```

Comment:
I don't know why but I feel like that check should just be `!!admin_mode ? thing_if_true : thing_if_false` no?

Reply:
TGUI is really finicky, that didn't seem to work with !!admin_mode || !!votable, so I had to do it that way